--- origsrc/libgcrypt-1.4.0/acinclude.m4	2007-07-26 04:08:53.000000000 -0500
+++ src/libgcrypt-1.4.0/acinclude.m4	2007-12-25 09:16:05.078125000 -0600
@@ -70,7 +70,7 @@
 AC_DEFUN([GNUPG_SYS_SYMBOL_UNDERSCORE],
 [tmp_do_check="no"
 case "${host}" in
-    *-mingw32msvc*)
+    *-mingw32msvc* | *-cygwin*)
         ac_cv_sys_symbol_underscore=yes
         ;;
     i386-emx-os2 | i[3456]86-pc-os2*emx | i386-pc-msdosdjgpp)
--- origsrc/libgcrypt-1.4.0/src/Makefile.am	2007-10-30 13:41:05.000000000 -0500
+++ src/libgcrypt-1.4.0/src/Makefile.am	2007-12-25 09:17:46.921875000 -0600
@@ -60,7 +60,6 @@
 gcrypt_res = versioninfo.lo
 gcrypt_res_ldflag = -Wl,.libs/versioninfo.o
 
-no_undefined = -no-undefined
 export_symbols = -export-symbols $(srcdir)/libgcrypt.def
 
 install-def-file:
@@ -75,7 +74,6 @@
 
 gcrypt_res =
 gcrypt_res_ldflag =
-no_undefined =
 export_symbols =
 install-def-file:
 uninstall-def-file:
@@ -85,7 +83,7 @@
 endif !HAVE_W32_SYSTEM
 
 
-libgcrypt_la_LDFLAGS = $(gcrypt_res_ldflag) $(no_undefined) $(export_symbols) \
+libgcrypt_la_LDFLAGS = $(gcrypt_res_ldflag) -no-undefined $(export_symbols) \
 	$(libgcrypt_version_script_cmd) -version-info \
 	@LIBGCRYPT_LT_CURRENT@:@LIBGCRYPT_LT_REVISION@:@LIBGCRYPT_LT_AGE@
 libgcrypt_la_DEPENDENCIES = ../cipher/libcipher.la ../mpi/libmpi.la \
